### PR TITLE
Feature/helpers tests

### PR DESCRIPTION
**Pull request by AlexandraMash**
### Description of Change ###

Added Unit tests + some comments for TextRange, TagsHelper and HashHelper classes

### API Changes ###

- TagsHelper. ExtractTags now throws ArgumentNullException for null input
- TagsHelper. ExtractTagsRanges now throws ArgumentNullException for null input
- TextRange constructor now throws ArgumentOutOfRangeException for negative parameters
- TextRange.BuildNewString now throws ArgumentNullException for null parameters
- TextRange ToString method was overriden

### Platforms Affected ### 

- Core (all platforms)

### PR Checklist ###

- [x] Targets the correct branch
- [x] Tests are passing (or failures are unrelated)

### CODE REVIEW

File: Softeq.XToolkit.Common.Tests/Helpers/TagsHelperTests/TagsHelperTests.cs [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+
+            Assert.Equal(1, result.Length);
+            Assert.Equal(textRange.Position, result[0].Position);
+            Assert.Equal(textRange.Length, result[0].Length);
+        }
+
+        [Fact]
+        public void ExtractTagsRanges_InputWithMultipleTags_ReturnsCorrectTagRange()
+        {
+            var input = TagsHelperDataProvider.InputWithMultipleTags;
```
Comment (wcoder) [inline Softeq.XToolkit.Common.Tests/Helpers/TagsHelperTests/TagsHelperTests.cs]: Can be used `ClassData` for arguments.

File: Softeq.XToolkit.Common/Helpers/TextRange.cs [hunk 2]
```diff
@@ -1,25 +1,62 @@
...
     public class TextRange
     {
+        /// <summary>
+        /// Creates a text range. Parameters can not be negative
+        /// </summary>
+        /// <param name="position">Start position of this range</param>
+        /// <param name="length">Length of this range</param>
         public TextRange(int position, int length)
         {
+            if (position < 0 || length < 0)
```
Comment (wcoder) [inline Softeq.XToolkit.Common/Helpers/TextRange.cs]: `ArgumentOutOfRangeException` can be better

File: Softeq.XToolkit.Common/Helpers/TextRange.cs [hunk 3]
```diff
@@ -1,25 +1,62 @@
...
+        /// Replaces the content of this text range with a new string
+        /// </summary>
+        /// <param name="fullString">Full string whose range will be altered</param>
+        /// <param name="newString">A new string to be inserted instead of the text range content</param>
+        /// <returns></returns>
+        public string BuildNewString(string fullString, string newString)
         {
-            return oldString
+            if (fullString == null || newString == null)
+            {
```
Comment (wcoder) [inline Softeq.XToolkit.Common/Helpers/TextRange.cs]: Can be better to split this check.

File: Softeq.XToolkit.Common/Helpers/TextRange.cs (hunk 2 above)
Comment (AlexandraMash) [reply Softeq.XToolkit.Common/Helpers/TextRange.cs]: Agree. Fixed

File: Softeq.XToolkit.Common/Helpers/TextRange.cs (hunk 3 above)
Comment (AlexandraMash) [reply Softeq.XToolkit.Common/Helpers/TextRange.cs]: You mean, to provide different messages?

File: Softeq.XToolkit.Common.Tests/Helpers/TagsHelperTests/TagsHelperTests.cs (hunk 1 above)
Comment (AlexandraMash) [reply Softeq.XToolkit.Common.Tests/Helpers/TagsHelperTests/TagsHelperTests.cs]: Done using MemberData

File: Softeq.XToolkit.Common/Helpers/TextRange.cs (hunk 3 above)
Comment (AlexandraMash) [reply Softeq.XToolkit.Common/Helpers/TextRange.cs]: Fixed

**Pull request merged**